skopt/utils.py

```diff
@@ -388,3 +387,61 @@ def point_aslist(search_space, point_as_dict):
     ]
     return point_as_list
 
+def normalize_dimensions(dimensions, estimator="GP"):
+    """
+    Normalize all dimensions.
+
+    This is particularly useful for Gaussian process based regressors and is used
+    internally by `gp_minimize.`
+
+    Parameters
+    ----------
+    * `dimensions` [list, shape=(n_dims,)]:
+        List of search space dimensions.
+        Each search dimension can be defined either as
+
+        - a `(upper_bound, lower_bound)` tuple (for `Real` or `Integer`
+          dimensions),
+        - a `(upper_bound, lower_bound, "prior")` tuple (for `Real`
+          dimensions),
+        - as a list of categories (for `Categorical` dimensions), or
+        - an instance of a `Dimension` object (`Real`, `Integer` or
+          `Categorical`).
+
+         NOTE: The upper and lower bounds are inclusive for `Integer`
+         dimensions.
+
+    * `estimator` ["GP", "RF", "ET", "GBRT", or sklearn regressor, default="GP"]:
+        optimization estimator to be used.
+        each estimator may use different transformations for dimensions.
+    """
+    if isinstance(estimator, str):
+        estimator = estimator.upper()
+        if estimator not in ["GP", "ET", "RF", "GBRT"]:
+            raise ValueError("Valid strings for estimator parameter "
+                             "are: 'RF', 'ET', 'GBRT', or 'GP', not %s" % estimator)
```

Comment:
I am stuck on this one. Travis is upset, and I am getting to following error:

![pytest_error](https://user-images.githubusercontent.com/16848331/27937968-e413dd88-6288-11e7-9da3-b8fa9565eb31.png)

Is my `ValueError` conflicting with the `@pytest.mark.fast_test`?